packages/picasso/src/LexicalEditor/LexicalEditor.tsx

```diff
@@ -163,45 +170,59 @@ const LexicalEditor = forwardRef<HTMLDivElement, Props>(function LexicalEditor(
   })
 
   return (
-    <LexicalComposer initialConfig={editorConfig}>
-      <div onFocus={handleFocus} onBlur={handleBlur} tabIndex={-1}>
-        <ToolbarPlugin
-          disabled={disabled || !isFocused}
-          toolbarRef={toolbarRef}
-          // remount Toolbar when disabled
-          key={`${disabled || !isFocused}`}
-        />
-        <OnChangePlugin ignoreSelectionChange onChange={handleChange} />
-        {autoFocus && <AutoFocusPlugin />}
-
-        <LexicalHeadingsReplacementPlugin />
-        <LexicalTextLengthPlugin onTextLengthChange={onTextLengthChange} />
-        <LexicalListPlugin />
-
-        <div className={classes.editorContainer} id={id} ref={ref}>
-          <RichTextPlugin
-            contentEditable={
-              <ContentEditable
-                className={classes.contentEditable}
-                data-testid={testIds?.editor}
+    <SettingsContext plugins={plugins}>
+      {({ nodes, settings, pluginsToRender }) => {
+        return (
+          <LexicalComposer
+            key={JSON.stringify(settings)}
+            initialConfig={editorConfig(nodes)}
+          >
+            <div onFocus={handleFocus} onBlur={handleBlur} tabIndex={-1}>
+              <ToolbarPlugin
+                disabled={disabled || !isFocused}
+                toolbarRef={toolbarRef}
+                // remount Toolbar when disabled
+                key={`${disabled || !isFocused}`}
+                customEmojis={customEmojis}
+              />
+              <OnChangePlugin ignoreSelectionChange onChange={handleChange} />
+              {autoFocus && <AutoFocusPlugin />}
+
+              <LexicalHeadingsReplacementPlugin />
+              <LexicalTextLengthPlugin
+                onTextLengthChange={onTextLengthChange}
               />
-            }
-            placeholder={
-              <Container
-                left='xsmall'
-                top='small'
-                className={classes.placeholder}
-              >
-                <Typography size='medium' color='grey-main-2'>
-                  {placeholder}
-                </Typography>
-              </Container>
-            }
-            ErrorBoundary={LexicalErrorBoundary}
-          />
-        </div>
-      </div>
-    </LexicalComposer>
+              <LexicalListPlugin />
+
+              {pluginsToRender.map((plugin: ReactNode) => plugin)}
```

Comment:
Why do we need this `.map` call? Do you mean to clone the array?
Maybe just
```suggestion
              {pluginsToRender}
```
wdyt?

Reply:
you are right, we probably don't need it

Reply:
why do we need a context for `pluginsToRender` ?

Reply:
as plugins will be mix of strings and ReactNodes, we need to split them and ReactNodes we want to just render, strings we need to initialize.

Reply:
as in this context we do logic around plugins, I thought it is the proper place to put it.